src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolBoundHandle.Unix.cs

```diff
@@ -8,8 +8,13 @@ namespace System.Threading
 {
     public sealed partial class ThreadPoolBoundHandle
     {
-        private static ThreadPoolBoundHandle BindHandleCore(SafeHandle handle)
+        private static ThreadPoolBoundHandle BindHandlePortableCore(SafeHandle handle)
         {
+            ArgumentNullException.ThrowIfNull(handle);
+
+            if (handle.IsClosed || handle.IsInvalid)
+                throw new ArgumentException(SR.Argument_InvalidHandle, nameof(handle));
+
             Debug.Assert(handle != null);
             Debug.Assert(!handle.IsClosed);
             Debug.Assert(!handle.IsInvalid);
```

Comment:
I don't think all this is necessary when it'll throw PNSE anyway

Reply:
Yea I recall that there was some test failing due to the exception here (expecting ArgumentException), I'll check more carefully as it shouldn't be necessary I guess

Reply:
Mono throws PNSE without checking arguments. If a test is failing for this API on Unix then something else may be going on.

Reply:
The assertions should be removed too